cloudpai/client.go

```diff
@@ -0,0 +1,117 @@
+package cloudpai
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/vektah/gqlparser/v2/gqlerror"
+
+	"time"
+)
+
+// defaultURL for Atlas Cloud.
+const defaultURL = "https://api.atlasgo.cloud/query"
+
+// roundTripper is a http.RoundTripper that adds the Authorization header.
+type roundTripper struct {
+	token string
+}
+
+// RoundTrip implements http.RoundTripper.
+func (r *roundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	req.Header.Set("Authorization", "Bearer "+r.token)
+	req.Header.Set("User-Agent", "gh-atlas-cli")
+	req.Header.Set("Content-Type", "application/json")
+	return http.DefaultTransport.RoundTrip(req)
+}
```

Comment:
I prefer unexported types to come after exported. Just a matter of style though.

Reply:
roundTripper is not exported though

Reply:
yes, this is why I like to have the code below the exported stuff. If I as a consumer visit the package I see what I can use first and don't have to scroll over loads of code I cannot access.